lib/async_hooks.js

```diff
@@ -476,6 +476,31 @@ function init(asyncId, type, triggerAsyncId, resource) {
   processing_hook = false;
 }
 
+// Integration with V8 Inspector AsyncTask* API
+
+const inspectorHook = createHook({
+  init(asyncId, type, triggerId, resource) {
+    // TODO(bajtos): support recurring tasks like setInterval
+    const recurring = false;
+    async_wrap.asyncTaskScheduled(type, asyncId, recurring);
+  },
+
+  before(asyncId) {
+    async_wrap.asyncTaskStarted(asyncId);
+  },
+
+  after(asyncId) {
+    async_wrap.asyncTaskFinished(asyncId);
+  },
+
+  destroy(asyncId) {
+    async_wrap.asyncTaskCanceled(asyncId);
+  },
+});
+
+// TODO(bajtos): enable the hook when Inspector is started,
+// disable it when Inspector is stopped
+inspectorHook.enable();
```

Comment:
@addaleax Thank you, this make sense. There is one missing piece - where to place the JS code glueing this all together. I think it should go to `lib/internal/bootstrap_node.js`, right?

Since the environment always have an instance of inspector `Agent`, I feel it's better to setup these two persistent Functions properties on `Agent` class instead. What do you think, can you foresee any issues with that?